tests/functional/codegen/modules/test_storage_variables.py

```diff
@@ -124,3 +124,50 @@ def get_lib_counter() -> uint256:
 
     assert c.counter() == 1
     assert c.get_lib_counter() == 5
+
+
+def test_import_complex_types(get_contract, make_input_bundle):
+    lib1 = """
+an_array: uint256[3]
+a_hashmap: HashMap[address, HashMap[uint256, uint256]]
+
+@internal
+def set_array_value(ix: uint256, new_value: uint256):
+    self.an_array[ix] = new_value
+
+@internal
```

Comment:
i mean i don't actually think it's that weird, external/internal is well-known to be the visibility rule, whether a function is exported in the abi. but yea it will feel more comfy when we make `@internal` optional.

Reply:
0.4 is a chance to change that expectation, and I think the context of ABI exporting is too confusing when you have to consider the perpindicularity of module access

with eliminating `@internal` and only keeping a marker for exporting a contract ABI item, it is clear what is actually driving contract interfaces and what is only just module code, to be used internal or externally by another module that initializes you